sql/binlog.cc

```diff
@@ -8955,6 +8960,60 @@ inline void MYSQL_BIN_LOG::update_binlog_end_pos(my_off_t pos) {
   unlock_binlog_end_pos();
 }
 
+my_off_t MYSQL_BIN_LOG::get_binlog_end_pos() const {
+  mysql_mutex_assert_not_owner(&LOCK_log);
+  return atomic_binlog_end_pos;
+}
+
+my_off_t MYSQL_BIN_LOG::get_last_acked_pos(bool &wait_for_ack, bool need_lock,
+                                           const char *sender_log_name) {
+  wait_for_ack = wait_for_ack && rpl_wait_for_semi_sync_ack &&
+                 rpl_semi_sync_master_enabled;
+
+  if (!wait_for_ack) return atomic_binlog_end_pos;
+
+  if (need_lock) lock_binlog_end_pos();
+  const int res = strcmp(last_acked.first.c_str(), sender_log_name);
```

Comment:
We should also check for file length here to compare binlog-1000 and binlog-999 correctly

Reply:
It's fixed now.